text/002-css-rules.md

```diff
@@ -0,0 +1,1094 @@
+      medium: { padding: 16 },
+      large: { padding: 24 }
+    }
+  },
+  // method 1
+  compoundVariants: [
+    {
+      color: brand,
+      size: small,
+      style: {
+        fontSize: "16px"
+      }
+    }
+  ],
+  // method 2
+  compoundVariants: ({ color, size }) => [
+    {
+      condition: [color.brand, size.small],
+      style: {
+        fontSize: "16px"
+      }
+    }
+  ]
+});
+```
+
+The first method is the original method in stitches's compound variants, and it is still a good way for it.  
+So we keep it.
+
+And the second method we present will provide a more comfortable way to write the conditions.  
+This feature checks existing variants and provides automatic completion.
+
+**Compiled:**
+
+```css
+.[FILE_NAME]_button_compound_0__[HASH] {
+  font-size: '16px'
+}
+.[FILE_NAME]_button_compound_1__[HASH] {
+  font-size: '24px',
+  font-weight: bold
+}
+```
+
+**Usage:**
+
+If you use the already set combination, it will be applied automatically.
+
+```typescript
+button({
+  color: "accent",
+  size: "large"
+});
+```
+
+## 7. Default Variants
+
+The way of [Stitches's Default Variants](https://stitches.dev/docs/variants#default-variants) is already good to use, so we keep this method in ours.
+
+**Code:**
+
+```typescript
+const button = rules({
+  // base styles
+
+  variants: {
+    color: {
+      brand: {
+        color: "#FFFFA0",
+        backgroundColor: "blueviolet"
+      },
+      accent: {
+        color: "#FFE4B5",
+        backgroundColor: "slateblue"
+      }
+    }
+  },
+  defaultVariants: {
+    color: "brand"
+  }
+});
+```
+
+## 8. Composition
+
+**Code:**
+
+[Vanilla Extract's Composition](https://vanilla-extract.style/documentation/style-composition/) is enough to use already. So we maintain this basically.  
+And It gonna inherit base rules like [Stitches's Composing Components](https://stitches.dev/docs/composing-components).
+
+```typescript
+const baseRule = rules({
+  padding: "12px",
+
+  variants: {
+    size: {
+      small: { fontSize: "11px" },
+      large: { fontSize: "20px" }
+    }
+  }
+});
+
+const myRule = rules([baseRule, { color: "blue", marginTop: "10px" }]);
+```
+
+**Compiled:**
+
+```css
+.[FILE_NAME]_baseRule__[HASH] {
+  padding: 12px;
+}
+
+.[FILE_NAME]_baseRule_size_small__[HASH] {
+  font-size: 11px;
+}
+.[FILE_NAME]_baseRule_size_large__[HASH] {
+  font-size: 20px;
+}
+
+.[FILE_NAME]_myRule__[HASH] {
+  color: "blue";
+  margin-top: "10px";
+}
+```
+
+## 9. rulesVarints
+
+Inspired by the [PandaCSS's `defineSlotRecipe()`](https://panda-css.com/docs/concepts/slot-recipes#config-slot-recipe).  
+And at the same time, We refer to [Vanilla Extract's `styleVariants`](https://vanilla-extract.style/documentation/api/style-variants/) for code style for now. It could be change to [sub functions](https://github.com/mincho-js/mincho/issues/59).
+
+This is a combination of `rules()` and `cssVariants()`.
+
+**Code:**
+
+```typescript
+const theme = rulesVariants({
```

Comment:
> [!NOTE]
> **P2[Strongly Recommended]: Please actively consider it. Changes are strongly encouraged, and if you don't make a change, you need a good reason.**

[`theme`](https://vanilla-extract.style/documentation/theming/) name, which may cause confusion with other cases in the future.

Instead, I recommend something that works for [BEM](https://getbem.com/)'s [case](https://medium.com/@YAKStack/bem-methodology-a-simple-and-effective-way-to-write-css-023607fb43b7).
![image](https://github.com/user-attachments/assets/18f39d72-b792-4556-b766-5b73f0f4a2d0)

- rulesVariants: Block
- Each Rules: Element
- Rule's variants: Modifier

Reply:
I missed this, it's my bad. I have updated docs based on ur suggestion.
The cases I wrote are different from ur example image but I followed BEM's case, of course.
If It's wrong, let me know again please. Thank you.